src/timeline/js/directives/ruler.js

```diff
@@ -57,14 +56,18 @@ App.directive("tlScrollableTracks", function () {
         // Send scrollbar position to Qt
         if (scope.Qt) {
            // Calculate scrollbar positions (left and right edge of scrollbar)
-           var timeline_length = Math.min(32767, scope.getTimelineWidth(0));
+           var timeline_length = scope.getTimelineWidth(0);
            var left_scrollbar_edge = scroll_left_pixels / timeline_length;
            var right_scrollbar_edge = (scroll_left_pixels + element.width()) / timeline_length;
 
            // Send normalized scrollbar positions to Qt
            timeline.ScrollbarChanged([left_scrollbar_edge, right_scrollbar_edge, timeline_length, element.width()]);
         }
 
+        scope.$apply( () => {
+          scope.scrollLeft = element[0].scrollLeft;
+        })
+
```

Comment:
This is probably a bad idea. Angular can always  just query the value directly from `$("#scrolling_tracks")` when it needs it. If a value that's meant to be ephemeral (like an element's current scroll position) gets copied and stored, the copy is almost instantly stale. Then you get issues like ZoomScroll's  janky dragging.

There are a few cases where you specifically want the value "at the time of the event". (Pointer coordinates are stored for mouse events because what matters is where the user clicked, _not_ where the pointer is now.) But for things like scroll position, keeping a copy of an old value is just inviting update lag.